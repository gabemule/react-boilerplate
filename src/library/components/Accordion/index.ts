import DsAccordion from "./Accordion"
import { DsAccordionItemProps }from "@/library/types/types";

export { DsAccordion }

export type { DsAccordionItemProps }
