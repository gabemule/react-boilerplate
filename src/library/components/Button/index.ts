import DsButton from "./Button";
import { DsButtonProps }from "./Button";

export { DsButton };
export type { DsButtonProps }
