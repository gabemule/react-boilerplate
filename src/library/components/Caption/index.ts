import DsCaption from "./Caption";
import { DsCaptionProps } from "./Caption"

export { DsCaption };
export type { DsCaptionProps }
