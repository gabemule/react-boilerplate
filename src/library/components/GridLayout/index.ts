import DsGridLayout from './GridLayout'
import DsGridElement from './GridElement'
import { DsGridLayoutProps } from './GridLayout'

export { DsGridLayout, DsGridElement }
export type { DsGridLayoutProps }
