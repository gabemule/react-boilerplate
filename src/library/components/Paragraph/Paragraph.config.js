module.exports = {
  name: 'Paragraph',
  class: 'ds-paragraph',
  props: {
    type: {
      type: String,
      default: 'small',
      options: ['large', 'small']
    }
  }
}
