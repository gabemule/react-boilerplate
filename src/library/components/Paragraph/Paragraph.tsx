import React from "react";
import withCustomProps from '@/library/decorators/withCustomProps'

// config
import ParagraphConfig from "./Paragraph.config";

export interface DsParagraphProps {
  children?: string,
  className?: string,
  type?: string,
}

const DsParagraph = ({
  children,
  className,
  type,
}: DsParagraphProps) => {
  const paragraphType = `${ParagraphConfig.class}--${type}`;

  return (
    <p
      data-testid="text-text"
      className={[ParagraphConfig.class, paragraphType, className && className].filter(classes => classes).join(" ")}>
      {children}
    </p>
  );
};

export default withCustomProps(DsParagraph, ParagraphConfig.props);
