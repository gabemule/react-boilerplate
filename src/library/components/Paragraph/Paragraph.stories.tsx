import React from "react";
import { Meta, Story } from "@storybook/react";

// config
import ParagraphConfig from "./Paragraph.config";

// components
import { ThemeProviderProps, ThemeProvider } from "@/library/components/ThemeProvider"
import { DsParagraph, DsParagraphProps } from "@/library/components/Paragraph";

const ComponentProps: Meta = {
  title: "Interação Básica/Typography/Paragraph",
  component: DsParagraph,
  argTypes: {
    type: {
      description: "Define o tamanho do texto",
      table: {
        category: "Style props",
        type: { summary: "string" },
      },
      options: ParagraphConfig.props.type.options,
      control: { type: "select" }
    },
  }
};

export default ComponentProps;

const Template: Story<ThemeProviderProps & DsParagraphProps> = (args) => (
  <ThemeProvider library={"core"} theme={"light"}>
    <DsParagraph {...args}>This is a Design System Text</DsParagraph>
  </ThemeProvider>
);

const defaultProps = {
  type: "small",
}

export const Default = Template.bind({});
Default.args = {
  ...defaultProps
}

export const Small = Template.bind({});
Small.args = {
  ...defaultProps
};

export const Large = Template.bind({});
Large.args = {
  ...defaultProps,
  type: "large",
};
