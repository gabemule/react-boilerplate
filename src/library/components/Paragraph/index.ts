import DsParagraph from "./Paragraph";
import { DsParagraphProps } from "./Paragraph";

export { DsParagraph };
export type { DsParagraphProps }
