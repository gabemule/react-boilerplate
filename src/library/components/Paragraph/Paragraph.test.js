import React from 'react'
import { shallow, render } from 'enzyme'

// components
import { DsParagraph } from '@/library/components/Paragraph'

// config
import ParagraphConfig from './Paragraph.config'

// components
const paragraphClass = ParagraphConfig.class
const paragraphPropText = 'Text Default Text'

describe('Text Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(
      <DsParagraph type={'large'}>{paragraphPropText}</DsParagraph>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = render(
      <DsParagraph type={'large'}>{paragraphPropText}</DsParagraph>
    )

    expect(wrapper).toBeTruthy()
    expect(wrapper.text()).toBe(paragraphPropText)
    expect(wrapper.find(paragraphClass)).toBeTruthy()
  })
})

describe('Props', () => {
  ParagraphConfig.props.type.options.forEach((type) => {
    it(`Accept prop type: ${type}`, () => {
      const wrapper = shallow(
        <DsParagraph type={type}>{paragraphPropText}</DsParagraph>
      )
      expect(wrapper.find(`${paragraphClass}--${type}`)).toBeTruthy()
    })
  })
})
