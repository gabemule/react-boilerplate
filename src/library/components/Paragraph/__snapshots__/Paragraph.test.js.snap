// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <p
    class="ds-paragraph ds-paragraph--large  with-custom-props"
    data-testid="text-text"
  >
    Text Default Text
  </p>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
